### PR TITLE
Add the MeshGenerator system

### PR DESCRIPTION
**Pull request by leorachapuis**
This is the new PR for the MeshGenerators (the old one was number #11640 ).
This should be nicer because we fixed some issues with old commits.

### CODE REVIEW

File: framework/doc/content/source/mesh_generators/AnnularMeshGenerator.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+The inner radius and the outer radius must be specified. If the inner radius is zero a disc mesh is created, while if it is positive an annulus is created. The annulus has just one subdomain (block number = 0), whereas the disc has two subdomains: subdomain zero consists of the outer quadrilaterals, while the other (block number = 1) consists of the triangular elements that eminate from the origin.
+
+The minimum and maximum angle may also be specified. These default to zero and $2\pi$, respectively. If other values are chosen, a sector of an annulus, or a sector of a disc will be created. Both angles are measured anticlockwise from the xx axis.
+
+The number of elements in the radial direction and the angular direction may be specified. In addition, a growth factor on the element size in the radial direction may be chosen. The element-size (in the radial direction) is multiplied by this factor for each concentric ring of elements, moving from the inner to the outer radius.
+
+Sidesets are also created:
```
Comment (friedmud) [inline framework/doc/content/source/mesh_generators/AnnularMeshGenerator.md]:
```suggestion
Sidesets created:
```

Comment (aeslaughter) [inline framework/doc/content/source/mesh_generators/AnnularMeshGenerator.md]:
```suggestion
Sidesets are created:
```

File: framework/doc/content/source/actions/AddMeshGeneratorAction.md [hunk 2]
```diff
@@ -0,0 +1,9 @@
+# AddMeshGeneratorAction
+
+!alert construction title=Undocumented Class
+The AddMeshGeneratorAction has not been documented, if you would like to contribute to MOOSE by writing
+documentation, please see [/generate.md]. The content contained on this page explains the typical
+documentation associated with an action; however, what is contained is ultimately determined by what
```
Comment (friedmud) [inline framework/doc/content/source/actions/AddMeshGeneratorAction.md]:
```suggestion
stuff however, what is contained is ultimately determined by what
```

**Comment by friedmud** [comment]
Started up many more of the optional tests - this adds so much code that I want to make sure it's tested well

**Comment by friedmud** [comment]
@permcody @andrsd @cpgr Whatever is wrong with RELAP-7's documentation has nothing to do with this PR.  I think I accidentally added that set of tests to this PR (I've now turned it off).

**Review by friedmud — DISMISSED** [review]
This is good - @leorachapuis has more work on this coming soon and we can change things over time too.

**Review by permcody — APPROVED** [review]

**Pull request merged**